src/index.ts

```diff
@@ -0,0 +1,347 @@
+/**
+ * Import styles library
+ */
+import styles from './styles.module.pcss';
+
+/**
+ * Import types
+ */
+import { NftToolData, NftToolConfig } from './types';
+import { API, BlockTool, PasteConfig, PatternPasteEvent } from '@editorjs/editorjs';
+import { IconNft } from './icons';
+import { IconChevronDown } from '@codexteam/icons';
+
+/**
+ * opensea Tool for Editor.js
+ */
+export default class NftTool implements BlockTool {
+  /**
+   * Code API — public methods to work with Editor
+   */
+  private readonly api: API;
+
+  /**
+   * Read-only mode flag
+   */
+  private readonly readOnly: boolean;
+
+  /**
+   * Tool data for input and output
+   */
+  private data: NftToolData;
+
+  /**
+   * Configuration object that passed through the initial Editor configuration.
+   */
+  private config: NftToolConfig;
+
+  /**
+   * Tool's HTML nodes
+   */
+  private nodes: {[key: string]: HTMLElement|null};
+
+  /**
+   * Class constructor
+   */
+  constructor({ data, config, api, readOnly }: { data: NftToolData, config: NftToolConfig, api: API, readOnly: boolean }) {
+    this.data = data;
+    this.config = config;
+    this.api = api;
+    this.readOnly = readOnly;
+
+    /**
+     * Declare Tool's nodes
+     */
+    this.nodes = {
+      wrapper: null,
+
+      image: null,
+      collection: null,
+      title: null,
+
+      refetchButton: null,
+    };
+  }
+
+  /**
+   * Add icon and title to the Toolbox
+   */
+  static get toolbox() {
+    return {
+      icon: IconNft,
+      title: 'Token Card'
+    };
+  }
+
+  /**
+   * Process pasted content before appending to the Editor
+   */
+  static get pasteConfig(): PasteConfig {
+    return {
+      patterns: {
+        opensea: NftTool.regexp,
+      }
+    };
+  }
+
+  /**
+   * Notify core that read-only mode is supported
+   */
+  static get isReadOnlySupported(): boolean {
+    return true;
+  }
+
+  /**
+   * Creates UI of a Block
+   */
+  render(): HTMLElement {
```

Comment:
Please add access modifiers to public methods
```suggestion
  public render(): HTMLElement {
```